kafka-connect/src/main/java/io/tabular/iceberg/connect/data/AlternateDynClasses.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.tabular.iceberg.connect.data;
+
+import java.util.Set;
+import org.apache.iceberg.relocated.com.google.common.base.Joiner;
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+
+/* Copied from  org.apache.iceberg.common.DynClasses.java
+  Due to wanting to avoid packaging Hadoop/Hive libraries in
+  versions of the connector that don't require it, we need
+  to adjust the class loader to handle an additional error
+*/
+public class AlternateDynClasses {
+
+  private AlternateDynClasses() {}
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+    private ClassLoader loader = Thread.currentThread().getContextClassLoader();
+    private Class<?> foundClass = null;
+    private boolean nullOk = false;
+    private Set<String> classNames = Sets.newLinkedHashSet();
+
+    private Builder() {}
+
+    /**
+     * Set the {@link ClassLoader} used to lookup classes by name.
+     *
+     * <p>If not set, the current thread's ClassLoader is used.
+     *
+     * @param newLoader a ClassLoader
+     * @return this Builder for method chaining
+     */
+    public Builder loader(ClassLoader newLoader) {
+      this.loader = newLoader;
+      return this;
+    }
+
+    /**
+     * Checks for an implementation of the class by name.
+     *
+     * @param className name of a class
+     * @return this Builder for method chaining
+     */
+    public Builder impl(String className) {
+      classNames.add(className);
+
+      if (foundClass != null) {
+        return this;
+      }
+
+      try {
+        this.foundClass = Class.forName(className, true, loader);
+      } catch (ClassNotFoundException e) {
+        // not the right implementation
+      } catch (NoClassDefFoundError e) {
```

Comment:
Without this 

```java.lang.NoClassDefFoundError: org/apache/hadoop/conf/Configuration
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:862)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:760)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:681)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:639)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:579)
	at org.apache.kafka.connect.runtime.isolation.DelegatingClassLoader.loadClass(DelegatingClassLoader.java:122)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:579)
	at org.apache.kafka.connect.runtime.isolation.PluginClassLoader.loadClass(PluginClassLoader.java:124)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at org.apache.iceberg.common.DynClasses$Builder.impl(DynClasses.java:68)
	at io.tabular.iceberg.connect.data.Utilities.loadHadoopConfig(Utilities.java:76)
	at io.tabular.iceberg.connect.data.Utilities.loadCatalog(Utilities.java:70)
	at io.tabular.iceberg.connect.IcebergSinkTask.open(IcebergSinkTask.java:82)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.openPartitions(WorkerSinkTask.java:660)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.access$1200(WorkerSinkTask.java:76)
```

because we strip hadoop out of the build in our gradle file.  So the runtime was compiled with it, but it's removed.  Hence `NoClassDefFoundError` rather than `ClassNotFoundException`.  

I don't think this should go into iceberg core, but we would need to bring this slightly adjusted class into the project like I have done here.